.github/workflows/cypress.yml

```diff
@@ -1,7 +1,8 @@
 name: End2End Testing with Cypress
 on:
-  push:
-    {}
+  pull_request:
+    branches:
+      - dev
```

Comment:
Isn't that the target branch? in our case main? See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-the-head-or-base-branch-of-a-pull-request

Reply:
Yes. I think we should run cypress on dev only (no need to run on master, since all changes should come via dev branch).

Reply:
However, as we allow direct pushes to dev, I think we should run it on a PR to main as well. Otherwise, we allow for untested code to be injected to main, don't we?

Reply:
Sure. :-)